packages/crashlytics/ios/RNFBCrashlytics/RNFBCrashlyticsInitProvider.m

```diff
@@ -79,26 +98,51 @@ + (BOOL)isCrashlyticsJavascriptExceptionHandlerChainingEnabled {
+  //
+  // Solution:
+  // 1) use the SDK component dependency graph specification infrastructure
+  // to declare the crashlytics dependency...
+  FIRDependency *crashlyticsDep =
+      [FIRDependency dependencyWithProtocol:@protocol(FIRCrashlyticsInstanceProvider)
+                                 isRequired:YES];
 
-/*
- * configureWithApp is automatically invoked by Firebase as this app is a registered FIRLibrary with
- * the SDK.
- *
- * At this point "configure" has already been called on the FIRApp instance. This behavior is meant
- * to mirror what is done in ReactNativeFirebaseCrashlyticsInitProvider.java
- *
- * This pattern may not be supported long term
- * https://github.com/firebase/firebase-ios-sdk/issues/2933
- */
-+ (void)configureWithApp:(FIRApp *)app {
-  // This setting is sticky. setCrashlyticsCollectionEnabled persists the setting to disk until it
-  // is explicitly set otherwise or the app is deleted. Jump to the setCrashlyticsCollectionEnabled
-  // definition to see implementation details.
-  [[FIRCrashlytics crashlytics]
-      setCrashlyticsCollectionEnabled:self.isCrashlyticsCollectionEnabled];
-  DLog(@"RNFBCrashlyticsInit initialization successful");
+  // 2) ...then we put the desired startup logic using the dependency in a creation block...
+  FIRComponentCreationBlock creationBlock =
+      ^id _Nullable(FIRComponentContainer *container, BOOL *isCacheable) {
+    if (!container.app.isDefaultApp) {
+      return nil;
+    }
+
+    // Ensure it's cached so it returns the same instance every time messaging is called.
+    *isCacheable = YES;
+
+    // This returns a FIRCrashlytics *, directly cast it later for practical use, reference:
+    // https://github.com/firebase/firebase-ios-sdk/blob/main/Crashlytics/Crashlytics/FIRCrashlytics.m#L282-L284
+    id<FIRCrashlyticsInstanceProvider> crashlytics =
+        FIR_COMPONENT(FIRCrashlyticsInstanceProvider, container);
+
+    // This setting is sticky. setCrashlyticsCollectionEnabled persists the setting to disk until it
+    // is explicitly set otherwise or the app is deleted. Jump to the
+    // setCrashlyticsCollectionEnabled definition to see implementation details.
+    [(FIRCrashlytics *)crashlytics
+        setCrashlyticsCollectionEnabled:self.isCrashlyticsCollectionEnabled];
+    // Check for this in logs to make sure this block executes during testing via
+    // xcrun simctl spawn booted log stream --level debug --style compact |grep RNFBCrashlyticsInit
+    DLog(@"RNFBCrashlyticsInit initialization successful");
+    return nil;
+  };
+
+  // 3) ...finally: register startup block to run w/Crashlytics dependency when ready
+  FIRComponent *crashlyticsInitProvider =
+      [FIRComponent componentWithProtocol:@protocol(RNFBCrashlyticsInitProviderProtocol)
+                      instantiationTiming:FIRInstantiationTimingEagerInDefaultApp
+                             dependencies:@[ crashlyticsDep ]
```

Comment:
```suggestion
                             dependencies:@[]
```

But, this API is changing in Firebase v11 to remove the array parameter, so this line will need to be removed in the near future. The updated API is staged in the release-11.0 branch: https://github.com/firebase/firebase-ios-sdk/blob/69b4dba1ada7ac5ccc73ca8351158224fc56d1ab/FirebaseCore/Extension/FIRComponent.h#L72 (fyi the podspec is staged on SpecsStaging)

Reply:
okay - thanks for the heads-up. I suppose I could put some dynamic responds-to-selector type logic here to check which implementation is in underlying to be forwards- + backwards-compatible but a compile error will make this fast enough to detect when v11 comes out, and our breaking changes policy states that we (react-native-firebase) expects control of firebase-ios-sdk version and we are able to do underlying breaks without surfacing them in our versions, so won't even trigger a breaking change at our layer.

TL;DR: going to defer this until compile error surfaces it then just change to match and release